### PR TITLE
feat: 4639 - E2E tests for share link

### PR DESCRIPTION
**Pull request by charles-testco**
## Reason for Change

- #4639

## Changes

- add
Tests for share link

## Notes for Reviewer
Tests are limited to Chrome and Edge only. FF does not allow access to clipboard

### CODE REVIEW

**Review by tihuan — COMMENTED** [review]
Thanks for the tests, @charles-testco ! LGTM 🎉 

I created a feedback PR with comments please kindly review and merge into this PR if it looks good to you!

https://github.com/chanzuckerberg/single-cell-data-portal/pull/4651

File: frontend/tests/utils/geneUtils.ts [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import { Page, expect } from "@playwright/test";
+import { ROUTES } from "src/common/constants/routes";
+import { TEST_URL } from "tests/common/constants";
+import { ADD_GENE_BTN, ADD_TISSUE_BTN } from "./constants";
+
+const FMG_EXCLUDE_TISSUES = ["blood"];
+const CELL_COUNT_ID = "cell-count";
+const CELL_TYPE_NAME_ID = "cell-type-name";
+const MARKER_GENE_BUTTON_ID = "marker-gene-button";
+const REGEX = /^\d+\.?\d{0,2}$/;
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/tests/utils/geneUtils.ts:10]: Renamed to TWO_DECIMAL_NUMBER_REGEX and moved to shared file

File: frontend/tests/features/wheresMyGene/shareLink.test.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+
+async function verifyShareLink(page: Page, tissues: string[], genes: string[]) {
+  // copy link to clipboard
+  const clipboardText: string = await page.evaluate(
+    "navigator.clipboard.readText()"
+  );
+
+  // verify link contains all tissues, genes and version number
+  const urlParams = new URLSearchParams(clipboardText);
+  urlParams.getAll("tissues").forEach((x) => expect(tissues).toContain(x));
```
[the remark this answers is not in the document]
Comment (charles-testco) [reply frontend/tests/features/wheresMyGene/shareLink.test.ts]: Resolved with feedback PR

**Comment by charles-testco** [comment]
@ashin-czi @atarashansky , pls review as I made a lot of changes to implement the additional tests

**Pull request merged**